src/resolve/resolved-bus.c

```diff
@@ -523,6 +523,7 @@ static int bus_method_resolve_hostname(sd_bus_message *message, void *userdata,
 
         q->bus_request = sd_bus_message_ref(message);
         q->request_family = family;
+        q->request_name = hostname;
```

Comment:
This is still not right - it's a string in the message, it will go out of scope once the message is removed. It needs to be strduped.

Reply:
I actually had strdup initially, but per an earlier comment in this PR, I just copied the pointer :

From @keszybz:
_A reference to message is kept, and hostname remains valid as long as message is alive. So there is no need to duplicate the string._

Personally, I think strdup seems more intuitive and would go that route. If we all agree, I'll add the strdup back again.

Reply:
no, leave it as it is now then, I didn't see that comment